// containers/comment-box/index.js
import React, {Component} from 'react'

import CommentList from 'containers/comment-list'
import CommentForm from 'containers/comment-form'

export default class CommentBox extends Component {
  maxId = 100

  state = {
    commentsArr: [
      this.createCommentItem('This is first test comment'),
      this.createCommentItem('This is second test comment')
    ]
  }

  createCommentItem(text) {
    return {
      text,
      id: this.maxId++,
      img: 'http://www.kichuraphoto.com/wp-content/uploads/2019/08/test.png'
    }
  }

  onCommentAdded = (text) => {
    console.log(text)
    const newComment = this.createCommentItem(text)
    this.setState(({commentsArr}) => {
      const newCommentsArr = [
        ...commentsArr,
        newComment
      ]
      return {
        commentsArr: newCommentsArr
      }
    })
  }

  render() {
    return (
      <div className="commentBox">
        <h1>Comments</h1>
        <CommentList comments={this.state.commentsArr}/>
        <CommentForm onCommentAdded={this.onCommentAdded}></CommentForm>
      </div>
    )
  }
}
